src/manifolds/SymmetricPositiveDefiniteGeneralizedBuresWasserstein.jl

```diff
@@ -0,0 +1,147 @@
+@doc raw"""
+    GeneralizedBurresWassertseinMetric{T<:AbstractMatrix} <: AbstractMetric
+
+The generalized Bures Wasserstein metric for symmetric positive definite matrices, see[^HanMishraJawanpuriaGao2021].
+
+This metric internally stores the symmetric positive definite matrix ``M`` to generalise the metric,
+where the name also follows the mentioned preprint.
+
+[^HanMishraJawanpuriaGao2021]:
+    > Han, A., Mishra, B., Jawanpuria, P., Gao, J.:
+    > _Generalized Bures-Wasserstein geometry for positive definite matrices_.
+    > arXiv: [2110.10464](https://arxiv.org/abs/2110.10464).
+"""
+struct GeneralizedBuresWassersteinMetric{T<:AbstractMatrix} <: RiemannianMetric
+    M::T
+    GeneralizedBuresWassersteinMetric(MM::TT) where {TT<:AbstractMatrix} = new{TT}(MM)
+end
+
+@doc raw"""
+    change_representer(M::MetricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric}, E::EuclideanMetric, p, X)
+
+Given a tangent vector ``X ∈ T_p\mathcal M`` representing a linear function on the tangent
+space at `p` with respect to the [`EuclideanMetric`](@ref) `g_E`,
+this is turned into the representer with respect to the (default) metric,
+the [`GeneralizedBuresWassersteinMetric`](@ref) on the [`SymmetricPositiveDefinite`](@ref) `M`.
+
+To be precise we are looking for ``Z∈T_p\mathcal P(n)`` such that for all ``Y∈T_p\mathcal P(n)```
+it holds
+
+```math
+⟨X,Y⟩ = \operatorname{tr}(XY) = ⟨Z,Y⟩_{\mathrm{BW}}
+```
+for all ``Y`` and hence we get ``Z = 2pXM + 2MXp``.
+"""
+change_representer(
+    ::MetricManifold{ℝ,SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric},
+    ::EuclideanMetric,
+    p,
+    X,
+)
+
+function change_representer!(
+    M::MetricManifold{ℝ,<:SymmetricPositiveDefinite,<:GeneralizedBuresWassersteinMetric},
+    Y,
+    ::EuclideanMetric,
+    p,
+    X,
+)
+    Y .= 2 .* (p * X * M.metric.M + M.metric.M * X * p)
+    return Y
+end
+
+@doc raw"""
+    distance(::MatricManifold{SymmetricPositiveDefinite,GeneralizedBuresWassersteinMetric}, p, q)
+
+Compute the distance with respect to the [`BuresWassersteinMetric`](@ref) on [`SymmetricPositiveDefinite`](@ref) matrices, i.e.
+
+```math
+    d(p.q) = \bigl(
+    \operatorname{tr}(M^{-1}p) + \operatorname{tr}(M^{-1}q)
+       - 2\operatorname{tr}\bigl( (p^{\frac{1}{2}M^{-1}qM^{-1}p^{\frac{1}[2}) \bigr)^\frac{1}[2} \Bigr),
```

Comment:
```suggestion
d(p,q) = \operatorname{tr}(M^{-1}p) + \operatorname{tr}(M^{-1}q)
       - 2\operatorname{tr}\bigl( (p^{\frac{1}{2}}M^{-1}qM^{-1}p^{\frac{1}{2}} \bigr)^{\frac{1}{2}},
```
The old equation didn't render correctly.